core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -205,7 +214,12 @@ public void onPositionDiscontinuity(int reason) {
         if (advertHasNotStarted()) {
             adGroupIndex = player.getCurrentAdGroupIndex();
             adIndexInGroup = player.getCurrentAdIndexInAdGroup();
-            notifyAdvertStart(advertBreaks.get(adGroupIndex));
+
+            if (canPlayAdverts(adGroupIndex)) {
```

Comment:
This should be the same as the `onTimelineChanged` code, it operates in the same way. We are always pairing the `start` and `end` events.